apps/blockchain/lib/blockchain/ethash.ex

```diff
@@ -251,7 +253,16 @@ defmodule Blockchain.Ethash do
       |> FNV.hash(Enum.at(mix, mix_index))
       |> Integer.mod(cache_size)
 
-    Enum.at(modified_cache, cache_index)
+    %{^cache_index => element} = uint32_cache
```

Comment:
Cool pattern match! Is this different than just `Map.fetch!()`?

Reply:
It's a good point. I prefer `Map.fetch!` because I think it's more legible in this case, but I was going with the pattern matching because I had read that [pattern matching was faster than fetch](https://medium.com/learn-elixir/speed-up-data-access-in-elixir-842617030514) (though I don't know how reliable those estimates are). But now that I'm looking back at that blog post, the difference was with `Map.fetch` not `Map.fetch!` because the former creates an extra tuple `{:ok, result}`. I'll dig a little bit more into it and see if there's a difference.

Reply:
Updated in https://github.com/mana-ethereum/mana/pull/670/commits/2fb5acfcb8b0e96d75662e12e880719bd1a69fd8. More legible and seems to be faster.